src/goInstallTools.ts

```diff
@@ -323,7 +330,15 @@ export function installTools(missing: string[], goVersion: SemVersion): Promise<
 	outputChannel.appendLine(''); // Blank line for spacing.
 
 	return missing.reduce((res: Promise<string[]>, tool: string) => {
-		return res.then(sofar => new Promise<string[]>((resolve, reject) => {
+		// Disable modules for staticcheck and gotests,
+		// which are installed with the "..." wildcard.
+		// TODO: ... will be supported in Go 1.13, so enable these tools to use modules then.
+		if (modulesOff || tool === "staticcheck" || tool === "gotests") {
+			envForTools['GO111MODULE'] = 'off';
+		} else {
+			envForTools['GO111MODULE'] = 'on';
```

Comment:
How does this affect tools that have not onboard their source code to use modules?

Reply:
It won't. You can still download tools that don't have `go.mod` files when you are in module mode.